shared/lib/files.js

```diff
@@ -120,13 +123,63 @@ async function getYaml(configPath) {
   return yaml.load(await readFile(configPath, 'utf8'), { json: true })
 }
 
+/**
+ * Compare arrays of FileSize types from the base and the head branch and add a
+ * percentage difference to the object. If there's no percentage difference,
+ * return a blank row.
+ *
+ * @param {FileSize[]} headFiles
+ * @param {FileSize[]} baseFiles
+ * @returns {(FileSize|null)[]} FileSize array with a percentage attribute, potentially reduced, or an empty array
+ */
+function getFileSizeComparison(headFiles, baseFiles) {
+  return headFiles
+    .map((file, i) => {
+      const headFileSize = file.size
+      const baseFileSize = baseFiles[i].size
```

Comment:
This assumes that the two files will have the same number of entries in the same order. This will likely break if we add, remove or rename files.

We'll need to `find` the matching file from `baseFiles` and handle cases where the file doesn't exist.

Ideally, we'd also handle deletions – this only takes into accounts files that exist in `HEAD` but we should show -100%.

I don't think we should worry about renames – let's accept that it'll appear as an addition + a deletion of 2 separate files.

Reply:
Additions and deletions are a good point. I've got some thoughts about looping over both arrays to produce a more accurate picture. Leave this with me.